x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/LogicalPlanBuilder.java

```diff
@@ -155,6 +158,15 @@ else if (!selectTarget.isEmpty()) {
         if (ctx.setQuantifier() != null && ctx.setQuantifier().DISTINCT() != null) {
             query = new Distinct(source(ctx.setQuantifier()), query);
         }
+
+        // TOP
+        SqlBaseParser.TopClauseContext topClauseContext = ctx.topClause();
+        if (topClauseContext != null) {
```

Comment:
Why not combining all three conditions in the same `if`?  More readable this way?